scrapy/spider.py

```diff
@@ -31,11 +31,19 @@ def __init__(self, name=None, **kwargs):
         if not hasattr(self, 'start_urls'):
             self.start_urls = []
 
-    def log(self, message, level=log.DEBUG, **kw):
-        """Log the given messages at the given log level. Always use this
-        method to send log messages from your spider
+    @property
+    def logger(self):
+        logger = logging.getLogger(self.name)
```

Comment:
hworker mostly works but there are some issues with backwards compatibility of `log.msg()`, see https://gist.github.com/dangra/ee2287a099680aade28a#file-log_loremipsum_2-txt-L23.